### DISCUSSION TITLE
Is it ok to register subscribers directly through the event manager?

### DISCUSSION THREAD

**Discussion by jbmikk** [mikro-orm/mikro-orm#1738] [Q&A]
The docs don't mention anything about this, but is it ok to manually add event subscribers like this:

    em.getEventManager().registerSubscriber(new MySubscriber());

Basically I have an EM factory in a DIC so I need to inject a different subscriber for each EM.

**Comment by B4nan** [comment]
Yes, although I am not sure what you mean by the last sentence about factories. You should only fork EMs, not create new ones. And a fork will share the same subscribers (the whole `EventManager` instance is shared across forks). Why you would want to have distinct subscribers per fork?
